### DISCUSSION TITLE
How to extract certain pages from PDF.

### DISCUSSION THREAD

**Discussion by ugoborghetti** [py-pdf/pypdf#1975] [Q&A]
I used this code inside my function to extract certain pages from a PDF. Now this is deprecated. How do I do this with pypdf? Or can someone point me to where this is explained in the documentation? I can't find anything.

    # Get the PDF file from the bucket
    bucket = client.get_bucket(BUCKET_NAME)
    blob = bucket.blob(PDF_FILE_NAME)
    pdf_bytes = io.BytesIO(blob.download_as_bytes())

    # Read the PDF and extract the desired pages
    pdf_reader = PyPDF2.PdfFileReader(pdf_bytes)
    pdf_writer = PyPDF2.PdfFileWriter()

    for page_number in PAGE_NUMBERS_TO_KEEP:
        page = pdf_reader.getPage(page_number - 1)
        pdf_writer.addPage(page)

    # Save the extracted pages to a new PDF file
    output_bytes = io.BytesIO()
    pdf_writer.write(output_bytes)

**Comment by pubpub-zz** [comment]
the better is to use `.append()` passing list of page numbers as `pages` argument
if multiple pages are coming from same document it is better to provide them all in a list to keep possible links and outlines

**Reply by pubpub-zz** [reply]
Rtfm🤓
https://pypdf.readthedocs.io/en/latest/user/merging-pdfs.html#append

**Reply by ugoborghetti** [reply]
I used the following [1] and it worked, but I get a strange output [2] from the console. Why is this?

[1]

```python
from pypdf import PdfWriter

# create a PdfWriter instance
writer = PdfWriter()

# append the first page of 'source.pdf' to the writer
writer.append("source.pdf", [0])

# write the output to 'output.pdf'
with open("output.pdf", "wb") as output_pdf:
    writer.write(output_pdf)
```

[2]
`(False, <_io.BufferedWriter name='output.pdf'>)`

**Reply by pubpub-zz** [reply]
this is normal:
False because you have pass a filename as a parameter 
<_io...> is the _repr_() of your output stream
